File: browserwn/src/androidTest/java/org/sqlunet/browser/wn/RunTextSearchTest.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.browser.wn;

import junit.framework.TestCase;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.sqlunet.browser.Seq;
import org.sqlunet.browser.MainActivity;

import androidx.annotation.NonNull;
import androidx.test.core.app.ActivityScenario;
import androidx.test.ext.junit.rules.ActivityScenarioRule;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.LargeTest;

@RunWith(AndroidJUnit4.class)
@LargeTest
public class RunTextSearchTest extends TestCase
{
	@NonNull
	@Rule
	public ActivityScenarioRule<MainActivity> activityScenarioRule = new ActivityScenarioRule<>(MainActivity.class);

	@Before
	public void before()
	{
		if (Do.ensureDownloaded())
			ActivityScenario.launch(MainActivity.class);

		Seq.do_navigate(R.id.drawer_layout, R.id.nav_view, R.id.nav_status);
		Do.ensureTextSearchSetup(R.id.searchtextWnButton);

		Seq.do_navigate(R.id.drawer_layout, R.id.nav_view, R.id.nav_search_text);
	}

	@Test
	public void searchWnDefinitionsRun()
	{
		Do.textSearchRun(0);
	}

	@Test
	public void searchWnSamplesRun()
	{
		Do.textSearchRun(1);
	}

	@Test
	public void searchWnWordsRun()
	{
		Do.textSearchRun(1);
	}
}

File: browsersn/src/androidTest/java/org/sqlunet/browser/sn/RunTextSearchTest.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser.sn

import androidx.test.core.app.ActivityScenario
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import junit.framework.TestCase
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.sqlunet.browser.MainActivity
import org.sqlunet.browser.Seq
import org.sqlunet.browser.sn.Do.ensureDownloaded
import org.sqlunet.browser.sn.Do.ensureTextSearchSetup
import org.sqlunet.browser.sn.Do.textSearchRun
import org.sqlunet.browser.sn.test.R

@RunWith(AndroidJUnit4::class)
@LargeTest
class RunTextSearchTest : TestCase() {

    @Rule
    var activityScenarioRule = ActivityScenarioRule(MainActivity::class.java)

    @Before
    fun before() {
        if (ensureDownloaded()) ActivityScenario.launch(MainActivity::class.java)
        Seq.doNavigate(R.id.drawer_layout, R.id.nav_view, R.id.nav_status)
        ensureTextSearchSetup(org.sqlunet.browser.sn.R.id.searchtextWnButton)
        Seq.doNavigate(R.id.drawer_layout, R.id.nav_view, R.id.nav_search_text)
    }

    @Test
    fun searchWnDefinitionsRun() {
        textSearchRun(0)
    }

    @Test
    fun searchWnSamplesRun() {
        textSearchRun(1)
    }

    @Test
    fun searchWnWordsRun() {
        textSearchRun(2)
    }
}